.github/actions/check-artifact-hash/action.yml

```diff
@@ -0,0 +1,56 @@
+name: Check the Git hash in an artifact if it exists
+
+inputs:
+  package_path:
+    description: 'The path to the package for which the artifact should be checked'
+    required: true
+    type: string
+
+outputs:
+  artifact_name:
+    description: "The name of the generated artifact"
+    value: ${{ steps.parse.outputs.artifact_name }}
+  needs_rebuild:
+    description: "Does the package need to be rebuilt?"
+    value: ${{ steps.compare.outputs.rebuild }}
+  hash:
+    description: "The hash to be saved as artifact"
+    value: ${{ steps.compare.outputs.hash }}
+
+runs:
+  using: composite
+  steps:
+    - id: parse
+      name: Parse artifact name
+      run: |
+        COMP_PATH=${{ inputs.package_path }}
+        COMP_PATH=${COMP_PATH//\//_}
+        echo "artifact_name=${COMP_PATH:2}" >> $GITHUB_OUTPUT
+      shell: bash
+
+    - uses: dawidd6/action-download-artifact@v2
```

Comment:
This action checks the generated artifacts of the latest successful workflow (build-release.yaml in our case) on a specific branch and downloads the artifact with name `name` if available